### PR TITLE
[FLINK-10842][E2E tests] fix broken waiting loops in common.sh

### PR DESCRIPTION
**Pull request by azagrebin**
## What is the purpose of the change

There are 3 loops in flink-end-to-end-tests/test-scripts/common.sh where the script waits for some event to happen (for i in {1..10}; do):

wait_dispatcher_running
start_and_wait_for_tm
wait_job_running

This PR changes them to fail if the expected event has not happened and to return from the loop if happened.

## Brief change log

  - change waiting loops in common.sh

## Verifying this change

Run any relevant e2e tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by azagrebin** [comment]
cc @zentol

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 1]
```diff
@@ -401,7 +406,7 @@ function wait_for_job_state_transition {
   echo "Waiting for job ($job) to switch from state ${initial_state} to state ${next_state} ..."
 
   while : ; do
-    N=$(grep -o "($job) switched from state ${initial_state} to ${next_state}" $FLINK_DIR/log/*standalonesession*.log | tail -1)
+    N=$(grep -o "($job) switched from state ${initial_state} to wait_dispatcher_running${next_state}" $FLINK_DIR/log/*standalonesession*.log | tail -1)
```
Comment (twalthr) [inline flink-end-to-end-tests/test-scripts/common.sh]: Is this change intended?

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 2]
```diff
@@ -412,17 +417,20 @@ function wait_for_job_state_transition {
 }
 
 function wait_job_running {
+  local TIMEOUT=10
   for i in {1..10}; do
```
Comment (twalthr) [inline flink-end-to-end-tests/test-scripts/common.sh]: Variable not used in loop.

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 3]
```diff
@@ -242,30 +245,45 @@ function start_taskmanagers {
...
+  wait_for_number_of_running_tms $((running_tms+1))
+}
 
-  for i in {1..10}; do
-    local new_running_tms=`curl ${CURL_SSL_ARGS} -s "${url}" | grep -o "id" | wc -l`
-    if [ $((new_running_tms-running_tms)) -eq 0 ]; then
-      echo "TaskManager is not yet up."
+function query_running_tms {
+  local url="${REST_PROTOCOL}://${NODENAME}:8081/taskmanagers"
+  curl ${CURL_SSL_ARGS} -s "${url}"
```
Comment (twalthr) [inline flink-end-to-end-tests/test-scripts/common.sh]: Could it be possible that this requests fails and crashes a script that has `set -e` enabled? Should we put a `|| true` at the end?

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 4]
```diff
@@ -242,30 +245,45 @@ function start_taskmanagers {
...
+}
+
+function query_number_of_running_tms {
+  query_running_tms | grep -o "id" | wc -l
```
Comment (twalthr) [inline flink-end-to-end-tests/test-scripts/common.sh]: Same concern here. `grep -o` can throw an error that should be caught.

File: flink-end-to-end-tests/test-scripts/common.sh (hunk 3 above)
Comment (azagrebin) [reply flink-end-to-end-tests/test-scripts/common.sh]: We check before (line 250) that cluster is running and responses correctly.
I would assume we do not expect this commands to fail.
If something is wrong with querying cluster, the script should fail fast, wdyt?

Comment (twalthr) [reply flink-end-to-end-tests/test-scripts/common.sh]: Yes, that makes sense to me. Thanks for the clarification.

**Comment by twalthr** [comment]
Thanks for the update @azagrebin. I will merge this...
